braindecode/models/eegsimpleconv.py

```diff
@@ -0,0 +1,151 @@
+
+    """
+
+    def __init__(
+        self,
+        # Base arguments
+        n_outputs=None,
+        n_chans=None,
+        sfreq=None,
+        # Model specific arguments
+        fm=128,
+        n_convs=2,
+        resampling=80,
+        kernel_size=8,
+        # Other ways to initialize the model
+        chs_info=None,
+        n_times=None,
+        input_window_seconds=None,
+    ):
+        super().__init__(
+            n_outputs=n_outputs,
+            n_chans=n_chans,
+            chs_info=chs_info,
+            n_times=n_times,
+            input_window_seconds=input_window_seconds,
+            sfreq=sfreq,
+        )
+        del n_outputs, n_chans, chs_info, n_times, sfreq, input_window_seconds
+
+        self.rs = (
+            Resample(orig_freq=self.sfreq, new_freq=resampling)
+            if self.sfreq != resampling
+            else torch.nn.Identity()
+        )
+
+        self.conv = torch.nn.Conv1d(
+            self.n_chans, fm, kernel_size=kernel_size,
+            padding=kernel_size // 2, bias=False
+        )
+        self.bn = torch.nn.BatchNorm1d(fm)
+        self.blocks = []
+        newfm = fm
+        oldfm = fm
+        for i in range(n_convs):
+            if i > 0:
+                newfm = int(1.414 * newfm)
+            self.blocks.append(
+                torch.nn.Sequential(
+                    (
+                        torch.nn.Conv1d(
+                            oldfm,
+                            newfm,
+                            kernel_size=kernel_size,
+                            padding=kernel_size // 2,
+                            bias=False,
+                        )
+                    ),
+                    (torch.nn.BatchNorm1d(newfm)),
+                    (torch.nn.MaxPool1d(2) if i > 0 - 1 else torch.nn.MaxPool1d(1)),
+                    (torch.nn.ReLU()),
+                    (
+                        torch.nn.Conv1d(
+                            newfm,
+                            newfm,
+                            kernel_size=kernel_size,
+                            padding=kernel_size // 2,
+                            bias=False,
+                        )
+                    ),
+                    (torch.nn.BatchNorm1d(newfm)),
+                    (torch.nn.ReLU()),
+                )
+            )
+            oldfm = newfm
+        self.blocks = torch.nn.ModuleList(self.blocks)
+        self.final_layer = torch.nn.Linear(oldfm, self.n_outputs)
+
+    def forward(self, x):
+        """
+        Forward pass of the model.
+
+        Parameters
+        ----------
+        x: PyTorch Tensor
+            Input tensor of shape (batch_size, n_channels, n_times)
+
+        Returns
+        -------
+        PyTorch Tensor
+            Output tensor of shape (batch_size, n_outputs)
+        """
+        x_rs = self.rs(x.contiguous())
+        y = torch.relu(self.bn(self.conv(x_rs)))
+        for seq in self.blocks:
+            y = seq(y)
+        y = y.mean(dim=2)
```

Comment:
I strongly recommend adding an extra variable to return the features if necessary.

Reply:
Done !